Source/MQTTnet/MqttClient.cs

```diff
@@ -1018,10 +1018,10 @@ async Task TrySendKeepAliveMessages(CancellationToken cancellationToken)
 
                 if (timeWithoutPacketSent > keepAlivePeriod)
                 {
-                    using (var timeoutCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken))
+                    using (var keepAliveCancellationTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken))
                     {
-                        timeoutCancellationTokenSource.CancelAfter(Options.Timeout);
-                        await PingAsync(timeoutCancellationTokenSource.Token).ConfigureAwait(false);
+                        keepAliveCancellationTokenSource.CancelAfter(keepAlivePeriod);
```

Comment:
With this change the cancellation will happen earlier according to the keep alive period (15 seconds). Which is by default way smaller than the generic communication timeout (100 seconds).

But we already reached _timeWithoutPacketSent > keepAlivePeriod_ so that wait an additional of 15 seconds (so 30 in total).

Isn't it a better way to wait for _keepAlivePeriod / 2_ to match the calculation at the server instead?